.ci/azure-pipelines/docs-pipeline.yaml

```diff
@@ -1,46 +1,33 @@
-trigger:
-  paths:
-    include:
-    - doc
+trigger: none
 
-pr:
-  paths:
-    include:
-    - doc
+pr: none
 
 resources:
   pipelines:
+    - pipeline: Main-CI
+      source: Main
+      trigger:
+        stages:
+        - documentation
     - pipeline: Build-CI
       source: Build
       trigger:
         stages:
         - build_gcc
   containers:
-    - container: fmt # for formatting.yaml
-      image: pointcloudlibrary/fmt
     - container: doc # for documentation.yaml
       image: pointcloudlibrary/doc
     - container: env1804 # for tutorials.yaml
       image: pointcloudlibrary/env:18.04
 
 stages:
-  - stage: formatting
-    displayName: Formatting
-    # if docs pipeline triggered by build_gcc stage,
-    # the formatting stage has already run, thus it
-    # won't run for a second time here.
-    condition: ne(variables['Build.Reason'], 'ResourceTrigger')
-    jobs:
-      - template: formatting.yaml
-
   - stage: documentation
     displayName: Documentation
-    condition: in(dependencies.formatting.result, 'Succeeded', 'SucceededWithIssues', 'Skipped')
     jobs:
       - template: documentation.yaml
 
   - stage: tutorials
     displayName: Tutorials
-    condition: in(dependencies.documentation.result, 'Succeeded', 'SucceededWithIssues')
+    dependsOn: documentation
```

Comment:
How does tutorials depends on documentation?

I would think that if there are code changes (including code changes in tutorials), then the build-CI should run and tutorials as well (its down stream projects that can catch install errors).

If there are only documentation changes, then only the documentation CI should run?

Reply:
Thank you for your reviewing! :grinning: 
> I would think that if there are code changes (including code changes in tutorials), then the build-CI should run and tutorials as well (its down stream projects that can catch install errors).

Sorry for ignoring the source codes in `pcl/doc/tutorials/content/sources/` in code changes diff. But I'm not very sure whether it's needed to run build-CI if there are only tutorials sources changed. 

I find that `tutorials` job is just to build tutorials source and report any errors from compiling (refered to https://github.com/PointCloudLibrary/pcl/issues/3594). Could we just assume tutorials changes are a part of documentation changes, and only run docs-CI (generate docs -> build tutorials)? If so, we could save the time of build-CI.

> If there are only documentation changes, then only the documentation CI should run?

If tutorials changes are processed only in docs-CI, we could also add `git-diff` at the beginning of tutorials stage, so that we can skip it if no tutorials source code changes.

Reply:
Yes, lets try starting with your suggestions.

Reply:
Thank you! I've finished exact that. Could you please take a look at this again? :grinning:

Reply:
Tutorial sources are a change of only tutorials and thus only the doc pipeline.

However, for the testing purposes, we build and install PCL before using that to build tutorials. So, tutorials need to be built if:
* PCL source code changed
* Tutorials source code changed

Reply:
Thanks for your review and the explanation! 

I'd change the tutorial diff stage to address the issue.